### PR TITLE
[#169167474] Added screen to enable NFC

### PR DESCRIPTION
**Pull request by Alex193a**
Added a screen to enable the NFC if the phone supports it but it is not enabled yet.

### CODE REVIEW

File: locales/it/index.yml [hunk 1]
```diff
@@ -224,6 +224,12 @@ authentication:
       noNfcConnectionContent: !include cie/noNfcConnection.md
     ok: Ok
     cieCardValid: Ok, la tua carta è valida! Attendi ancora qualche secondo..
+  cie:
+    enableNfcTitle: Attiva NFC
+    enableNfcContent: Per permettere all'app di verificare la tua carta d'identità elettronica devi prima attivare la funzione NFC del tuuo dispositivo. Torna qui quando avrai finito!
```
Comment (valentinacala) [inline locales/it/index.yml]:
```suggestion
    enableNfcContent: Per permettere all'app di verificare la tua Carta di Identità Elettronica devi prima attivare la funzione NFC del tuo dispositivo. Torna qui quando avrai finito!
```

File: locales/it/index.yml [hunk 2]
```diff
@@ -787,3 +793,4 @@ reminders:
   loadMoreData: carica le scadenze del periodo precedente
 openMaps:
   genericError: Si è verificato un errore durante l'apertura della mappa
+genericError: Si è verificato un errore
```
Comment (valentinacala) [inline locales/it/index.yml]: You could shift this alert under `general`

File: ts/screens/authentication/NfcStartScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import * as React from "react";
+import { Linking, StyleSheet } from "react-native";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+import BaseScreenComponent from "../../components/screens/BaseScreenComponent";
+import FooterWithButtons from "../../components/ui/FooterWithButtons";
+import I18n from "../../i18n";
+import ROUTES from "../../navigation/routes";
+import variables from "../../theme/variables";
+
+type OwnProps = Readonly<{
```
Comment (valentinacala) [inline ts/screens/authentication/NfcStartScreen.tsx]:
```suggestion
type Props = Readonly<{
```

File: ts/screens/authentication/NfcStartScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      <Container>
+        <BaseScreenComponent goBack={true}>
+          <View style={styles.contentContainerStyle}>
+            <H1>{I18n.t("authentication.cie.enableNfcTitle")}</H1>
+            <View spacer={true} />
+            <Text style={styles.text}>
+              {I18n.t("authentication.cie.enableNfcContent")}
+            </Text>
+            <View spacer={true} />
+            <Text link={true} onPress={() => this.browseToLink()}>
```
Comment (valentinacala) [inline ts/screens/authentication/NfcStartScreen.tsx]: It should work:
```suggestion
            <Text link={true} onPress={this.browseToLink}>
```

File: ts/screens/authentication/NfcStartScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+              {I18n.t("authentication.cie.enableNfcHelp")}
+            </Text>
+          </View>
+        </BaseScreenComponent>
+        <FooterWithButtons
+          type="TwoButtonsInlineThird"
+          leftButton={{
+            cancel: true,
+            onPress: (): boolean => this.props.navigation.goBack(),
```
Comment (valentinacala) [inline ts/screens/authentication/NfcStartScreen.tsx]: Can you check if it works?
```suggestion
            onPress: this.props.navigation.goBack,
```

Comment (Alex193a) [reply ts/screens/authentication/NfcStartScreen.tsx]: Yes, it works!

File: locales/it/index.yml (hunk 2 above)
Comment (Alex193a) [reply locales/it/index.yml]: I can't find `general`, do you mean `global`?

File: locales/en/index.yml [hunk 6]
```diff
@@ -268,7 +274,7 @@ authentication:
       title: Login error
       body: There was an error with the login, please retry!
     spid:
-      unknown: Unknown error 
+      unknown: Unknown error
```
[the remark this answers is not in the document]
Comment (Alex193a) [reply locales/en/index.yml]: I can't find `general`, do you mean `global`?

File: locales/it/index.yml (hunk 2 above)
Comment (valentinacala) [reply locales/it/index.yml]: yes!

Comment (Alex193a) [reply locales/it/index.yml]: Ok, genericError is already under global

File: ts/screens/authentication/NfcActiveScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            <Text style={styles.text}>
+              {I18n.t("authentication.cie.nfcEnabledContent")}
+            </Text>
+          </View>
+        </BaseScreenComponent>
+        <FooterWithButtons
+          type="TwoButtonsInlineThird"
+          leftButton={{
+            cancel: true,
+            onPress: (): boolean => this.props.navigation.goBack(),
```
Comment (valentinacala) [inline ts/screens/authentication/NfcActiveScreen.tsx]:
```suggestion
            onPress: this.props.navigation.goBack,
```

File: locales/en/index.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (valentinacala) [reply locales/en/index.yml]: Yes!

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#169167474](https://www.pivotaltracker.com/story/show/169167474): Come CIT, se NFC è supportato ma non è attivo, alla pressione del pulsante "entra con CIE" voglio vedere un screen che mi invita ad attivarlo

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/screens/authentication/NfcActiveScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+            icon={require("../../../img/icons/nfc-icon.png")}
+          />
+          <View style={styles.contentContainerStyle}>
+            <Text style={styles.text}>
+              {I18n.t("authentication.cie.nfcEnabledContent")}
+            </Text>
+          </View>
+        </BaseScreenComponent>
+        <FooterWithButtons
+          type="TwoButtonsInlineThird"
```
Comment (valentinacala) [inline ts/screens/authentication/NfcActiveScreen.tsx]:
```suggestion
          type={"TwoButtonsInlineThird"}
```

File: ts/screens/authentication/NfcActiveScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+          leftButton={{
+            cancel: true,
+            onPress: (): void => {
+              this.props.navigation.goBack();
+            },
```
Comment (valentinacala) [inline ts/screens/authentication/NfcActiveScreen.tsx]: it should be reduced to `onPress: this.props.navigation.goBack`

File: ts/screens/authentication/NfcActiveScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+            title: I18n.t("global.buttons.cancel"),
+            block: true
+          }}
+          rightButton={{
+            cancel: false,
+            onPress: (): void => Alert.alert(I18n.t("global.notImplemented")),
```
Comment (valentinacala) [inline ts/screens/authentication/NfcActiveScreen.tsx]: You can add here a comment related to the story implementing this this feature: 
`// TODO: proceed with authentication https://www.pivotaltracker.com/story/show/169685147`

p.s. Usually for onPress you can omit the output type of the function
```suggestion
            onPress: () => Alert.alert(I18n.t("global.notImplemented")),
```

File: ts/screens/authentication/NfcStartScreen.tsx [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+            >
+              {I18n.t("authentication.cie.enableNfcHelp")}
+            </Text>
+          </View>
+        </BaseScreenComponent>
+        <FooterWithButtons
+          type="TwoButtonsInlineThird"
+          leftButton={{
+            cancel: true,
+            onPress: (): void => {
```
Comment (valentinacala) [inline ts/screens/authentication/NfcStartScreen.tsx]: `onPress: this.props.navigate.goBack` should work also here. It avoids we create a new instance of the function each time the onPress is called (happening when arrow functions are used)

File: ts/screens/authentication/NfcStartScreen.tsx [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+                  ROUTES.AUTHENTICATION_SPID_INFORMATION
+                )
+              }
+            >
+              {I18n.t("authentication.cie.enableNfcHelp")}
+            </Text>
+          </View>
+        </BaseScreenComponent>
+        <FooterWithButtons
+          type="TwoButtonsInlineThird"
```
Comment (valentinacala) [inline ts/screens/authentication/NfcStartScreen.tsx]:
```suggestion
          type={"TwoButtonsInlineThird"}
```

File: ts/screens/authentication/NfcStartScreen.tsx [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+          <View style={styles.contentContainerStyle}>
+            <Text style={styles.text}>
+              {I18n.t("authentication.cie.enableNfcContent")}
+            </Text>
+            <View spacer={true} />
+            <Text
+              link={true}
+              onPress={(): boolean =>
+                this.props.navigation.navigate(
+                  ROUTES.AUTHENTICATION_SPID_INFORMATION
```
Comment (valentinacala) [inline ts/screens/authentication/NfcStartScreen.tsx]: It links to the SPID info screen while it should be related to the CIE. If it is not available on master, we could wait it is available and point to the proper screen.

File: ts/screens/authentication/LandingScreen.tsx [hunk 14]
```diff
@@ -69,7 +69,9 @@ const cardProps: ReadonlyArray<ComponentProps<typeof LandingCardComponent>> = [
 const LandingScreen: React.SFC<Props> = props => {
   const navigateToMarkdown = () => props.navigation.navigate(ROUTES.MARKDOWN);
   const navigateToIdpSelection = () =>
-    props.navigation.navigate(ROUTES.AUTHENTICATION_IDP_SELECTION);
+    config.isCIEauthenticationEnabled
```
Comment (Undermaken) [inline ts/screens/authentication/LandingScreen.tsx]: I guess this could be removed

File: ts/screens/authentication/NfcStartScreen.tsx [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+          type={"TwoButtonsInlineThird"}
+          leftButton={{
+            cancel: true,
+            onPress: this.props.navigation.goBack,
+            title: I18n.t("global.buttons.cancel"),
+            block: true
+          }}
+          rightButton={{
+            cancel: false,
+            onPress: (): boolean =>
```
Comment (Undermaken) [inline ts/screens/authentication/NfcStartScreen.tsx]: Avoid arrow function.
Make it as a class method and pass only the reference

```typescript
private handlePressContinue = () => {
    // TODO: use CIE-module to open the NFC-OS settings page and then refresh the status 
    // (if nfc is enabled or not) to 
    // handle the navigation properly
    this.props.navigation.navigate(ROUTES.AUTHENTICATION_CIE_NFC_ENABLED);
  };
[...]
onPress: this.handlePressContinue,
```

**Pull request merged**